meta-mender-core/recipes-mender/mender-client/files/mender-client-resize-data-part.sh.in

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+set -e
+
+# All sizes are in blocks of 512 bytes
+
+MENDER_STORAGE_DEVICE=@MENDER_STORAGE_DEVICE@
+mender_storage_device_base=$(basename ${MENDER_STORAGE_DEVICE})
+
+# 32 MB, overstimate here and should be close enough
+PARTITION_OVERHEAD=65536
+
+total_disk_size=$(cat /sys/block/${mender_storage_device_base}/size)
+calculated_size=0
+
+for part in $(find /sys/block/${mender_storage_device_base}/ -name ${mender_storage_device_base}* -exec basename {} \;); do
```

Comment:
`${mender_storage_device_base}*` should be quoted, it's quite prone to mis-expansion due to the wildcard.